scripts/west_commands/footprint.py

```diff
@@ -0,0 +1,80 @@
+# Copyright (c) 2020 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+
+import argparse
+import os
+import pprint
+import re
+from pathlib import Path
+import subprocess
+
+'''west "footprint" command'''
+
+from west.commands import WestCommand
+from runners.core import BuildConfiguration
+from build_helpers import find_build_dir, is_zephyr_build, \
+    FIND_BUILD_DIR_DESCRIPTION
+from run_common import desc_common, add_parser_common, do_run_common
+
+from zcmake import CMakeCache
+from zephyr_ext_common import Forceable, cached_runner_config, \
+    zephyr_scripts_path
+
+
+class Footprint(Forceable):
+
+    def __init__(self):
+        super(Footprint, self).__init__(
+            'footprint',
+            # Keep this in sync with the string in west-commands.yml.
+            'show footprint information of a binary.',
+            'Show binary code, data, stack usage.',
+            accepts_unknown_args=True)
+
+    def do_add_parser(self, parser_adder):
+        parser = parser_adder.add_parser(
+            self.name,
+            help=self.help,
+            formatter_class=argparse.RawDescriptionHelpFormatter,
+            description=self.description)
+
+        parser.add_argument('-d', '--build-dir',
+                            help=FIND_BUILD_DIR_DESCRIPTION)
+        self.add_force_arg(parser)
+
+        return parser
+
+    def do_run(self, args, runner_args):
+        self.args = args        # for check_force
+
+        # Find the build directory and parse .config and DT.
+        build_dir = find_build_dir(args.build_dir)
+        self.check_force(os.path.isdir(build_dir),
+                         'no such build directory {}'.format(build_dir))
+        self.check_force(is_zephyr_build(build_dir),
+                         "build directory {} doesn't look like a Zephyr build "
+                         'directory'.format(build_dir))
+
+        bcfg = BuildConfiguration(build_dir)
+        zephyr_base = Path(os.environ.get('ZEPHYR_BASE',
+                       Path(__file__).parent.parent.parent))
+
+        cache = CMakeCache.from_build_dir(build_dir)
+        compiler_base = cache.get('CMAKE_C_COMPILER')
+
+        if compiler_base:
+            compiler_base = re.sub("gcc$", "", compiler_base)
+
+        runner_config = cached_runner_config(build_dir, cache)
+        cmd = ['puncover',
```

Comment:
I suggest we either revive the `puncover` project or fork it officially and publish it in pypi under our zephyr account as `zephyr-puncover` or similar. Thoughts?

Reply:
Hey, author of puncover here. @carlescufi , can you let me know which areas of puncover need care and are seemingly dead? I am willing to spend time and merge feature requests (as done over the last days) to avoid forks.

Reply:
@HBehrens Thanks for the comment! when I posted my comment it was Python2-only and had not seen commits for a while. I see this has changed now, and so I see no reason to fork it or move it elsewhere. Would you consider publishing `puncover` to pypi so we can add it to our `requirements.txt`?

Reply:
Thanks, I created [issue 34 at the puncover project](https://github.com/HBehrens/puncover/issues/34) to capture this.